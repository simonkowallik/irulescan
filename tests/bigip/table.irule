when CLIENT_ACCEPTED priority 500 {
    log local0. "table set -- foo1 bar1> [table  set -- foo1 bar1]"
    log local0. "table set -subtable foo3 -- abc def> [table  set -subtable foo3 -- abc def]"
    log local0. "table set -georedundancy -- abc def> [table  set -georedundancy -- abc def]"
    log local0. "table set -notouch -subtable foo4 -- abc def> [table  set -notouch -subtable foo4 -- abc def]"
    log local0. "table add -- foo2 bar2> [table  add -- foo2 bar2]"
    log local0. "table replace -- foo1 1> [table  replace -- foo1 1]"
    log local0. "\[table lookup -- foo1\]> [table lookup -- foo1]"
    log local0. "\[table incr -- foo1 1\]> [table incr -- foo1 1]"
    log local0. "\[table lookup -georedundancy -- abc\]> [table lookup -georedundancy -- abc]"
    log local0. "table append -- foo2 bar3> [table  append -- foo2 bar3]"
    log local0. "table delete -- foo2> [table  delete -- foo2]"
    log local0. "table timeout -- foo1 60> [table  timeout -- foo1 60]"
    log local0. "table lifetime -- foo1 60> [table  lifetime -- foo1 60]"
    log local0. "\[table timeout -remaining -- foo1\]> [table timeout -remaining -- foo1]"
    log local0. "\[table lifetime -remaining -- foo1\]> [table lifetime -remaining -- foo1]"
    log local0. "\[table keys -subtable foo3 -count\]> [table keys -subtable foo3 -count]"
}
when CLIENT_ACCEPTED priority 501 {
    log local0. "table set FOO1 bar1> [table  set FOO1 bar1]"
    log local0. "table set -subtable FOO3 ABC DEF> [table  set -subtable FOO3 ABC DEF]"
    log local0. "table set -georedundancy ABC DEF> [table  set -georedundancy ABC DEF]"
    log local0. "table set -notouch -subtable FOO4 ABC DEF> [table  set -notouch -subtable FOO4 ABC DEF]"
    log local0. "table add FOO2 bar2> [table  add FOO2 bar2]"
    log local0. "table replace FOO1 1> [table  replace FOO1 1]"
    log local0. "\[table lookup FOO1\]> [table lookup FOO1]"
    log local0. "\[table incr FOO1 1\]> [table incr FOO1 1]"
    log local0. "\[table lookup -georedundancy ABC\]> [table lookup -georedundancy ABC]"
    log local0. "table append FOO2 bar3> [table  append FOO2 bar3]"
    log local0. "table delete FOO2> [table  delete FOO2]"
    log local0. "table timeout FOO1 60> [table  timeout FOO1 60]"
    log local0. "table lifetime FOO1 60> [table  lifetime FOO1 60]"
    log local0. "\[table timeout -remaining FOO1\]> [table timeout -remaining FOO1]"
    log local0. "\[table lifetime -remaining FOO1\]> [table lifetime -remaining FOO1]"
    log local0. "\[table keys -subtable FOO3 -count\]> [table keys -subtable FOO3 -count]"
}