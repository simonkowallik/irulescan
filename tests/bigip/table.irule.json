[{"filepath":"table.irule.json","warning":[],"dangerous":[]}]
