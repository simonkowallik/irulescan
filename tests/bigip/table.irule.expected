DANGEROUS: missing options terminator `--` permits argument injection at `FOO1` in `table  set FOO1 bar1`
DANGEROUS: missing options terminator `--` permits argument injection at `ABC` in `table  set -subtable FOO3 ABC DEF`
DANGEROUS: missing options terminator `--` permits argument injection at `ABC` in `table  set -georedundancy ABC DEF`
DANGEROUS: missing options terminator `--` permits argument injection at `ABC` in `table  set -notouch -subtable FOO4 ABC DEF`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO2` in `table  add FOO2 bar2`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO1` in `table  replace FOO1 1`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO1` in `table lookup FOO1`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO1` in `table incr FOO1 1`
DANGEROUS: missing options terminator `--` permits argument injection at `ABC` in `table lookup -georedundancy ABC`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO2` in `table  append FOO2 bar3`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO2` in `table  delete FOO2`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO1` in `table  timeout FOO1 60`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO1` in `table  lifetime FOO1 60`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO1` in `table timeout -remaining FOO1`
DANGEROUS: missing options terminator `--` permits argument injection at `FOO1` in `table lifetime -remaining FOO1`

